sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -70,17 +74,17 @@ public void execute(SensorContext context) {
     }
   }
 
-  private static boolean shouldExecuteOnProject(FileSystem fs) {
-    if (!SystemUtils.IS_OS_WINDOWS) {
-      LOG.debug("OS is not Windows. Skip Sensor.");
-      return false;
-    }
-
+  private boolean shouldExecuteOnProject(FileSystem fs) {
     if (!filesToAnalyze(fs).iterator().hasNext()) {
       LOG.debug("No files to analyze. Skip Sensor.");
       return false;
     }
 
+    if (!system.isOsWindows()) {
```

Comment:
We are checking the OS after we check if there are files to analyze in order to allow the user to disable the SonarC# plugin when they are analyzing on non-Windows OS projects that contain C# files (for some reason)